releasenotes/notes/add-null_value-matrix-generator-1225e7d553fe89a3.yaml

```diff
@@ -0,0 +1,31 @@
+---
+features:
+  - |
+    The :func:`~retworkx.adjacency_matrix` function has a new kwarg
+    ``null_value`` which is used to adjust the value used in the output matrix
+    representing the absense of an edge. This can be set to any float value and
+    if not specified the default value of 0.0 is still used. For example:
```

Comment:
What happens with existing edges? Is it always `1.0`?

I image the adj matrix would either be bools (0/1) or weights; if weights then `inf` would make more sense as a default. And below, zero length existing path seems valid to me (especially when weights can be floats).

It seems this PR would be the first step towards changing the default. Is that part of the plan?

Reply:
The adjacency matrix uses the edge weight for the value (and the sum of edge weights for parallel edges) as returned by the user specified callback weight function (or if that's not specified there is a default weight value https://qiskit.org/documentation/retworkx/stubs/retworkx.adjacency_matrix.html#retworkx.adjacency_matrix lists the existing options) for the elements in the matrix where there is an edge . Previously the default was to just use 0.0 for places where there was no edge the new option lets us use any float instead of 0.0 (including `inf` or `nan`) for elements where this no edge

Reply:
I don't have any plan to change the default, more just to make it configurable for people that don't want to use 0.0 to indicate the lack of an edge. There is a potential conflict if the weight function returns 0.0 so letting people have the flexibility to pick makes sense. I'm adding the dual of this to `from_adjacency_matrix` in #409